baseapp/baseapp.go

```diff
@@ -636,7 +639,7 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte, tx sdk.Tx) (result sdk
 	}
 
 	ctx = ctx.WithMultiStore(msCache)
-	result = app.runMsgs(ctx, msgs)
+	result = app.runMsgs(ctx, msgs, mode)
```

Comment:
why bother passing in the mode ? do we need to run this method at all on checktx?

Reply:
We do check that the message could have been routed (that there was a handler), which seems worth doing.